### ISSUE TITLE
DeepID2

### ISSUE THREAD

**Issue by denghuiru** [happynear/FaceVerification#8]
@happynear 
Hi, I have researched the face verification for a period of time and keep learning DeepID recently. I found that your implement the DeepID2 prototxt in "mnist_siamese_train_test". However, I have a problem that need your help.
1. Firstly, how do you prepare the datasets(Webface), I am not very clear about your webface_pair1_train and webface_pair2_train and webface_pair1_test and webface_pair2_test, how do you partiton the dataset and why you have two training sets and test sets.
Thanks very much!

**Comment by denghuiru** [comment]
@happynear 
Hi, if it is possible, will you send the datasets to me that used for training DeepID2.

**Comment by happynear** [comment]
I haven't trained DeepID2 successfully. The loss cannot converge. However, you can still try by yourself.

Data partition scripts are stored in `dataset` folder. You can use `getTrainList.m` to get the pair list. There is a parameter `same_r = 0.5` refering to the  ratio of same pairs. The script is not well designed. It would crash some times, so you can run it again if error.

Besides, I have also provided a `get10TrainList.m` to get 10 times more pairs. You can try it. Since it would cost too much time creating such a big database, you'd better use ImageDataLayer and SSD hard disk.

**Comment by denghuiru** [comment]
@happynear 
oh, Thanks you very much for these advices, I will try this model firstly. Besides, have you ever trained DeepID only with verification information, if you have some experience, please give me some instructions.

**Comment by denghuiru** [comment]
@happynear 
Hi, I still have a problem, when you train DeepID2 model, how you combine verification and identification information together, it seems that your manner is different from the paper said, you separately get the softmaxLoss output, and then used it in the ContrastiveLossLayer, is it proper to combine in this way? why you don't try to do as the paper described.

**Comment by happynear** [comment]
The margin of contrastive loss is difficult to set. I suggest you to use only same pairs.

**Comment by happynear** [comment]
https://github.com/happynear/FaceVerification/blob/master/caffe_proto/mnist_siamese_train_test_concate.prototxt
This file is exactly followed the paper. Simultaneously optimize softmax and contrastive loss.

**Comment by denghuiru** [comment]
@happynear 
Thank you very much, I will deeply see this DeepID2 model, and try to implement it. By the way, Does this model can only be used with your windows-version caffe?
Have you ever trained this model yet?

**Comment by happynear** [comment]
I've said before, the net cannot converge. However, you can still try it.
